alacritty_terminal/src/config/mod.rs

```diff
@@ -242,20 +243,47 @@ impl Default for EscapeChars {
 }
 
 #[serde(default)]
-#[derive(Deserialize, Copy, Clone, Debug, Default, PartialEq, Eq)]
+#[derive(Deserialize, Copy, Clone, Debug, PartialEq)]
 pub struct Cursor {
     #[serde(deserialize_with = "failure_default")]
     pub style: CursorStyle,
     #[serde(deserialize_with = "option_explicit_none")]
     pub vi_mode_style: Option<CursorStyle>,
+    #[serde(deserialize_with = "default_thickness")]
+    thickness: Percentage,
     #[serde(deserialize_with = "failure_default")]
     unfocused_hollow: DefaultTrueBool,
 }
 
 impl Cursor {
+    #[inline]
     pub fn unfocused_hollow(self) -> bool {
         self.unfocused_hollow.0
     }
+
+    #[inline]
+    pub fn thickness(self) -> f64 {
+        self.thickness.0 as f64
+    }
+}
+
+impl Default for Cursor {
+    fn default() -> Self {
+        Self {
+            style: Default::default(),
+            vi_mode_style: Default::default(),
+            thickness: Percentage::new(DEFAULT_CURSOR_THICKNESS),
+            unfocused_hollow: Default::default(),
+        }
+    }
+}
+
+pub fn default_thickness<'a, D>(deserializer: D) -> Result<Percentage, D::Error>
```

Comment:
The naming here isn't really great, since usually that's more the naming we use for default values, instead of deserializers.

So it should be something like `deserialize_cursor_thickness`.